lib/typing/ext/parse.go

```diff
@@ -62,48 +48,27 @@ func ParseTimeExactMatchLegacy(layout, potentialDateTimeString string) (time.Tim
 // 1) Precision loss in translation
 // 2) Original format preservation (with tz locale).
 // If it cannot find it, then it will give you the next best thing.
-func ParseExtendedDateTime(dtString string, additionalDateFormats []string) (*ExtendedTime, error) {
-	// Check all the timestamp formats
-	var potentialFormat string
-	var potentialTime time.Time
+func ParseExtendedDateTime(val string, additionalDateFormats []string) (*ExtendedTime, error) {
+	// TODO: ExtendedTimeKindType so we can selectively parse.
 	for _, supportedDateTimeLayout := range supportedDateTimeLayouts {
-		ts, exactMatch, err := ParseTimeExactMatchLegacy(supportedDateTimeLayout, dtString)
-		if err == nil {
-			potentialFormat = supportedDateTimeLayout
-			potentialTime = ts
-			if exactMatch {
-				return NewExtendedTime(ts, DateTimeKindType, supportedDateTimeLayout), nil
-			}
+		if ts, err := ParseTimeExactMatch(supportedDateTimeLayout, val); err == nil {
+			return NewExtendedTime(ts, DateTimeKindType, supportedDateTimeLayout), nil
 		}
 	}
 
 	// Now check DATE formats, btw you can append nil arrays
 	for _, supportedDateFormat := range append(supportedDateFormats, additionalDateFormats...) {
-		ts, exactMatch, err := ParseTimeExactMatchLegacy(supportedDateFormat, dtString)
-		if err == nil && exactMatch {
+		if ts, err := ParseTimeExactMatch(supportedDateFormat, val); err == nil {
 			return NewExtendedTime(ts, DateKindType, supportedDateFormat), nil
 		}
 	}
 
-	// TODO: Remove this if we don't see any Sentry.
 	// Now check TIME formats
 	for _, supportedTimeFormat := range SupportedTimeFormatsLegacy {
-		ts, exactMatch, err := ParseTimeExactMatchLegacy(supportedTimeFormat, dtString)
-		if err == nil && exactMatch {
-			slog.Error("Unexpected call to SupportedTimeFormatsLegacy",
```

Comment:
Should we remove this entire loop? Looks like we don't actually expect to match any time formats, but with this change we'll still support matching time formats.

Reply:
We should keep it because we'll need it for the inference work later (When we encounter a `TIME` column later and we have a string value, we should try to parse the string value as `TIME`)